src/main/java/com/google/googleidentity/oauth2/token/InMemoryOAuth2TokenService.java

```diff
@@ -0,0 +1,451 @@
+            }
+        }
+
+        private String decryptTokenString() {
+            try {
+                Cipher cipher = Cipher.getInstance("AES");
+                byte[] bytesToDecrypt = BaseEncoding.base64Url().decode(tokenString);
+                cipher.init(Cipher.DECRYPT_MODE, aesKey);
+                byte[] decryptedBytes = cipher.doFinal(bytesToDecrypt);
+                return new String(decryptedBytes, "UTF-8");
+            } catch (Exception e) {
+                log.log(Level.INFO, "Error when encode tokenString!",  e);
+                return null;
+            }
+        }
+
+    }
+
+    private void initKey() {
+        KeyGenerator generator = null;
+        try {
+            generator = KeyGenerator.getInstance("AES");
+            generator.init(256);
+        } catch (NoSuchAlgorithmException e) {
+            log.log(Level.INFO, "Error when init Key!",  e);
+            return;
+        }
+        aesKey = generator.generateKey();
+    }
+
+    private class TokenCleaner implements Runnable {
+
+        @Override
+        public void run() {
+            for (Map.Entry<String, HashMap<String, HashMap<String, OAuth2AccessToken>>> userMap :
+                    userClientAccessTokenMap.entrySet()) {
+                for (Map.Entry<String, HashMap<String, OAuth2AccessToken>> tokenMap :
+                        userMap.getValue().entrySet()) {
+                    for (Map.Entry<String, OAuth2AccessToken> token :
+                            tokenMap.getValue().entrySet()) {
+                        if (token.getValue().getExpiredTime() < (new Date()).getTime()/1000l) {
+                            tokenMap.getValue().remove(token.getKey());
+                            if (tokenMap.getValue().isEmpty()) {
+                                userMap.getValue().remove(tokenMap.getKey());
+                                if (userMap.getValue().isEmpty()) {
+                                    userClientAccessTokenMap.remove(userMap.getKey());
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    };
+
+    private void setTokenCleaner() {
+        service = Executors.newSingleThreadScheduledExecutor();
+        service.scheduleAtFixedRate(
+                new TokenCleaner(), 60*60, 60*60, TimeUnit.SECONDS);
+    }
+
+    public OAuth2AccessToken generateAccessToken(OAuth2Request request) {
+
+        String clientID = request.getRequestAuth().getClientId();
+        String username = request.getRequestAuth().getUsername();
+
+        if (request.getRequestBody().getRefreshable()) {
+            if (!userClientRefreshTokenMap.containsKey(username)) {
+                 userClientRefreshTokenMap.put(username, new HashMap<>());
+            }
+
+            HashMap<String, OAuth2RefreshToken>  refreshTokenMap=
+                    userClientRefreshTokenMap.get(username);
```

Comment:
Rather than directly managing maps all over, it'd be better to add some semantically meaningful helper methods to get and store tokens.

This is also somewhat covered by my comment above to consider introducing some classes to manage users and their tokens.